scripts/irods/test/test_igroupadmin.py

```diff
@@ -72,6 +71,63 @@ def tearDownClass(self):
+        try:
+            # Use the groupadmin account to create group and set itself up as administrator.
+            self.groupadmin.assert_icommand( ['igroupadmin', 'mkgroup', self.group ], 'EMPTY' )
+            self.groupadmin.assert_icommand( ['igroupadmin', 'atg', self.group, self.groupadmin.username ], 'EMPTY' )
+
+            # Another groupadmin is needed for this test.
+            self.groupadmin2 = session.mkuser_and_return_session('groupadmin', 'groupadmin2', 'rods', lib.get_hostname())
+
+            # Test another user (a rodsuser) can be created and added to the group by this.
+            self.groupadmin.assert_icommand( ['igroupadmin', 'mkuser', self.rodsuser_name], 'EMPTY' )
+            self.groupadmin.assert_icommand( ['igroupadmin', 'atg', self.group, self.rodsuser_name], 'EMPTY' )
+
+            user_with_zone = lambda user_name,zn='': '{}#{}'.format(user_name, zn if zn else self.groupadmin.zone_name)
+
+            # Test that the groupadmin itself, and the new rodsuser, are members of the group.
+            self.groupadmin.assert_icommand(['igroupadmin', 'lg', self.group, ], 'STDOUT_MULTILINE', [user_with_zone(self.rodsuser_name),
+                                                                                                       user_with_zone(self.groupadmin.username)])
+
+            # Test removal of users, and self, from a group, as well as assignment of a new admin for the group via "atg".
+            self.groupadmin.assert_icommand(['igroupadmin', 'rfg', self.group, self.rodsuser_name], 'EMPTY')         # remove rodsuser
+            self.groupadmin.assert_icommand(['igroupadmin', 'atg', self.group, self.groupadmin2.username], 'EMPTY')  # add other groupadmin
+            self.groupadmin.assert_icommand(['igroupadmin', 'rfg', self.group, self.groupadmin.username], 'EMPTY')   # remove self
+
+            # Test that groupadmin2 is now a member of the group, and that groupadmin and the rodsuser have successfully been removed.
+            rc, out, _ = self.groupadmin2.assert_icommand(['igroupadmin', 'lg', self.group], 'STDOUT_MULTILINE', user_with_zone(self.groupadmin2.username))
+            self.assertEqual(rc, 0)
+            self.assertNotIn(user_with_zone(self.rodsuser_name), out)
+            self.assertNotIn(user_with_zone(self.groupadmin.username), out)
+
+            # Test that the new groupadmin can do some necessary things.
+            self.groupadmin2.assert_icommand(['igroupadmin', 'atg', self.group, self.rodsuser_name], 'EMPTY', desired_rc = 0)
+            self.groupadmin2.assert_icommand(['igroupadmin', 'rfg', self.group, self.rodsuser_name], 'EMPTY', desired_rc = 0)
+        finally:
+            self.admin.run_icommand(['iadmin', 'rmgroup', self.group])
+            self.groupadmin2.__exit__()
+            self.admin.assert_icommand(['iadmin', 'rmuser', self.groupadmin2.username])
+            self.admin.run_icommand(['iadmin', 'rmgroup', self.rodsuser_name])
```

Comment:
Do we need to remove the `groupadmin2` from the group before removing the actual group?

Reply:
The groupadmin2 account needs to be removed, yes, along with the session_collection, and that's done in the __exit__ call ... but I think it's immaterial the order in which it's done.

Reply:
You're saying `__exit__()` handles the session collection, not the removal of the groupadmin user from the group, correct?

> but I think it's immaterial the order in which it's done.

The order of what exactly?

Reply:
Sorry, I was thinking you were referring to a different line, so I had the wrong concept in my head...  I think you are right, I don't need to remove the groupadmin2 account; I can just delete the group and the DB handles it.  I'll change the test.

Reply:
Well, you need to remove the user from the group before doing anything else.

Reply:
Do you? I think I just did it without needing to....
```
irods@precision-5530:~/scripts$ iadmin mkgroup lab
irods@precision-5530:~/scripts$ iadmin atg lab dan
irods@precision-5530:~/scripts$ iadmin rmgroup lab
irods@precision-5530:~/scripts$ 
```

Reply:
Confirmed, I can delete L129 above with the redundant groupadmin2 removal, and the test still passes.

Reply:
No, nevermind that. error on  my part , that doesn't remove the user from the group, it removes the user. Maybe we can get on Zoom and I can get the sense of what your original question was, because now I'm kind of confused.